src/Arrays/vector_operations.jl

```diff
@@ -378,3 +378,30 @@ A scalar representing ``‖ x - y ‖_p``.
 function distance(x::AbstractVector, y::AbstractVector, p::Real=2.0)
     return norm(x - y, p)
 end
+
+"""
+    allequal(x)
+
+Check whether all elements in a sequence are equal
+
+### Input
+
+- `x` -- sequence
+
+### Output
+
+`true` iff all elements in `x` are equal.
+
+### Notes
+
+The code is taken from [here](https://stackoverflow.com/a/47578613).
+"""
+function allequal(x)
```

Comment:
`all(==(first(x)), x)` works too

Reply:
Yes, in the link there was a benchmark and that was slower I think.

Reply:
Yes, in the link there was a benchmark and that was slower I think.